include/aws/common/json.h

```diff
@@ -0,0 +1,325 @@
+
+/**
+ * Gets the string of a string aws_json_value.
+ * @param node The string aws_json_value.
+ * @return The string of the string aws_json_value, otherwise NULL.
+ */
+AWS_COMMON_API
+int aws_json_value_get_string(const struct aws_json_value *node, struct aws_byte_cursor *output);
+
+/**
+ * Gets the number of a number aws_json_value.
+ * @param node The number aws_json_value.
+ * @return The number of the number aws_json_value, otherwise NULL.
+ */
+AWS_COMMON_API
+int aws_json_value_get_number(const struct aws_json_value *node, double *output);
+
+/**
+ * Gets the boolean of a boolean aws_json_value.
+ * @param node The boolean aws_json_value.
+ * @return The boolean of the boolean aws_json_value, otherwise false.
+ */
+AWS_COMMON_API
+int aws_json_value_get_boolean(const struct aws_json_value *node, bool *output);
+// ====================
+
+// ====================
+// Object API
+
+/**
+ * Adds a aws_json_value to a object aws_json_value.
+ * @param object The object aws_json_value you want to add a node to.
+ * @param key The key to add the aws_json_value at.
+ * @param node The aws_json_value you want to add.
+ * @return AWS_OP_SUCCESS if adding was successful, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_add(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                        const struct aws_allocator *allocator, const struct aws_json_value *node);
+
+/**
+ * Returns the aws_json_value at the given key.
+ * @param object The object aws_json_value you want to get the node from.
+ * @param key The key that the aws_json_value is at. Is case sensitive.
+ * @return The aws_json_value at the given key, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_object_get(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                                           const struct aws_allocator *allocator);
+
+/**
+ * Returns the aws_json_value at the given case insensitive key.
+ * @param object The object aws_json_value you want to get the node from.
+ * @param key The key that the aws_json_value is at. Is not case sensitive.
+ * @return The aws_json_value at the given key, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_object_get_insensitive(const struct aws_json_value *object,
+                                                       const struct aws_byte_cursor *cursor,
+                                                       const struct aws_allocator *allocator);
+
+/**
+ * Checks if there is a aws_json_value at the given key.
+ * @param object The object aws_json_value you want to check a key in.
+ * @param key The key that you want to check. Is case sensitive.
+ * @return AWS_OP_SUCCESS if a aws_json_value is found, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_has(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                        const struct aws_allocator *allocator);
```

Comment:
since we don't actually create an aws_json_value struct that stores its allocator, we could have these getters use whatever allocator was passed to initialize cjson